packages/fast-usdc/src/util/noble.js

```diff
@@ -0,0 +1,98 @@
+import { DirectSecp256k1HdWallet, Registry } from '@cosmjs/proto-signing';
+import { AminoTypes, SigningStargateClient } from '@cosmjs/stargate';
+import { nobleAminoConverters, nobleProtoRegistry } from '@nick134-bit/noblejs';
+import axios from 'axios';
+
+const createMsgRegisterAccount = (
+  /** @type {string} */ signer,
+  /** @type {string} */ recipient,
+  /** @type {string} */ channel,
+) => {
+  return {
+    typeUrl: '/noble.forwarding.v1.MsgRegisterAccount',
+    value: {
+      signer,
+      recipient,
+      channel,
+    },
+  };
+};
+
+export const registerFwdAccount = async (
+  /** @type {string} */ nobleSeed,
+  /** @type {string} */ nobleToAgoricChannel,
+  /** @type {string} */ nobleRpc,
+  /** @type {string} */ recipient,
+  out = console,
+) => {
+  out.log('registering fwd account on noble');
+  const wallet = await DirectSecp256k1HdWallet.fromMnemonic(nobleSeed, {
+    prefix: 'noble',
+  });
+  out.log('got noble wallet from seed');
+  const accounts = await wallet.getAccounts();
+  const signerAddress = accounts[0].address;
+  out.log(`got signer ${signerAddress}`);
+  const msg = createMsgRegisterAccount(
+    signerAddress,
+    recipient,
+    nobleToAgoricChannel,
+  );
+  const fee = {
+    amount: [
+      {
+        denom: 'uusdc',
+        amount: '20000',
+      },
+    ],
+    gas: '200000',
+  };
+  out.log('signing message', msg);
+  const clientWithSigner = await SigningStargateClient.connectWithSigner(
+    nobleRpc,
+    wallet,
+    {
+      aminoTypes: new AminoTypes({
+        ...nobleAminoConverters,
+      }),
+      registry: new Registry([...nobleProtoRegistry]),
+    },
+  );
+  const txResult = await clientWithSigner.signAndBroadcast(
+    signerAddress,
+    [msg],
+    fee,
+    'Register Account Transaction',
+  );
+
+  if (txResult.code !== undefined && txResult.code !== 0) {
+    throw new Error(
+      `Transaction failed with code ${txResult.code}: ${txResult.events || ''}`,
+    );
+  }
+  return `Transaction successful with hash: ${txResult.transactionHash}`;
+};
+
+export const queryForwardingAccount = async (
+  /** @type {string} */ nobleApi,
+  /** @type {string} */ nobleToAgoricChannel,
+  /** @type {string} */ agoricAddr,
+  out = console,
+  get = axios.get,
```

Comment:
**note** (something I hope to do after this PR):

While this does support OCap discipline by letting the caller override the `get` parameter, `get` has excess authority: it's access to the whole web. [makeLCD](https://github.com/Agoric/agoric-sdk/blob/9ffdc26a7466d0ac9ad786dae07922e0d1984ec8/packages/fast-usdc/src/cli/cosmos-api.js#L16) combines `fetch` and an API url to make something with all and only the authority to use the Cosmos REST API at one endpoint. Here's hoping we manage to factor that out in due course.

Reply:
Ack